src/redux/store/reducer.js

```diff
@@ -13,7 +13,7 @@ const myReducer = (state = initialState, action) => {
   if(action.type === 'ADD_ALBUMS_ASYNC'){
     newState.albumList = action.albums;
   }
-  if(action.tyoe === 'CLEAR_ALBUMS'){
+  if(action.type === 'CLEAR_ALBUMS'){
     newState.albumList = [];
   }
 
```

Comment:
* For should be replaced with javascript functions like (forEach, filter, ...)
* You can manage to set state in ADD_IMAGE_ASYNC in sags, you don't need to iterate the Array again. (Each iteration cost you time)
* Don't forget to do ...state for property in the state which should stay unchanged for each action